app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def update_posts_count
+    author.increment!(:posts_counter)
+  end
+
+  def recent_five_comments
+    comments.order(created_at: :desc).limit(5)
+  end
```

Comment:
- [Optional] Impressive job @Talha-Hanif5666. My suggestion for you is to make use of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) to make your implementation even better. You can make the `update_post_counter` method private and call it with `after_save :update_post_counter`. This is a nice and clear implementation. 
- [Optional] This same approach could be used fro `update_comments_counter` and `update_likes_counter` methods.